test/state/requests.hpp

```diff
@@ -8,10 +8,14 @@
 #include "transaction.hpp"
 #include <evmc/evmc.hpp>
 #include <span>
-#include <vector>
 
 namespace evmone::state
 {
+/// The address of the deposit contract.
+///
+/// TODO: This address differs in different chains, so it should be configurable.
+constexpr auto DEPOSIT_CONTRACT_ADDRESS = 0x00000000219ab540356cBB839Cbe05303d7705Fa_address;
```

Comment:
Would it be possible to add some kind of an assertion whenever that constant is accessed, e.g. check the chain_id? This TODO is likely to be missed.

Reply:
It depends how the tests will evolve. It seems this should be a parameter of the JSON tests. In any case, this requires some research. As long as we pass all tests, this is not priority.

Reply:
I only meant in the unlikely event someone integrates evmone into a full client and forgets this.

Reply:
> integrates evmone into a full client and forgets
should have been "integrates before we fix the TODO"

Reply:
Yes, this is likely to happen. We had similar issue in Silkworm where we had chain id hardcoded to 1. This was only noticed by syncing a testnet.